### PR TITLE
Major cleanup of testing & documentation

### LINKED ISSUE #123: Run latest cookstyle
### :speaking_head: Foreword

Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

Chef have released updated cookstyle rules, we should therefore run the auto fix against the cookbook

1. Ensure you are on the latest stable `chef-workstation`
1. Run `cookstyle -a`

Issue sprouted from sous-chefs/meta/issues/111. If not applicable then issue should be closed.

### PR DESCRIPTION
**Pull request by detjensrobert**
### Description

- Adds full InSpec testing for Kitchen suite
- Adds complete documentation for resources
- Refactors ChefSpec tests for full coverage
- Refactors test recipes and suites
- Idempotency fixes for test recipes
- Updates old conventions to modern standards

### Breaking Changes
- Removed support for:
  - Amazon Linux
  - Amazon Linux 2
  - Debian
  - Ubuntu

### Issues Resolved

Fixes #123 #121 

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

-----

Signed-off-by: Robert Detjens <detjensrobert@osuosl.org>

### CODE REVIEW

**Comment by detjensrobert** [comment]
SELinux testing does not work with Dokken / Docker and breaks GHCI due to a reboot needed to fully enable SEL. Should the GHCI Dokken testing be removed or disabled?

**Comment by ramereth** [comment]
> SELinux testing does not work with Dokken / Docker. Should the GHCI Dokken testing be removed or disabled?

The only way to do this is via some runner which supports Virtualbox+Vagrant. Unfortunately GH Actions [don't support nested virtualization yet](https://github.com/actions/virtual-environments/issues/183) so that's not really an option. The other option is doing this from a [self hosted runner](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) but I doubt Chef Inc would want to do that.

Perhaps the best option is to just remove the dokken ci portion of this and also remove the dokken yaml in the meantime and force use of Vagrant? I'd want to see what @tas50 thinks however.

File: kitchen.yml [hunk 1]
```diff
@@ -8,6 +8,12 @@ provisioner:
   chef_license: accept-no-persist
   enforce_idempotency: true
   multiple_converge: 2
+  # reboot tweaks:
+  max_retries: 3
+  wait_for_retry: 90
+  retry_on_exit_code: [35, 213]
+  client_rb:
+    client_fork: false
```
Comment (ramereth) [inline kitchen.yml:16]: As I noted elsewhere offline, this doesn't seem to be working with CentOS 8 or Fedora.

Comment (ramereth) [reply kitchen.yml:16]: This was resolved with some additional changes.

**Review by ramereth — APPROVED** [review]
LGTM

**Pull request closed without merging**